src/planner/ompl/Planner.cpp

```diff
@@ -372,6 +374,81 @@ trajectory::InterpolatedPtr planCRRTConnect(
       std::move(_interpolator),
       _maxPlanTime);
 }
+
+//=============================================================================
+std::pair <std::unique_ptr<trajectory::Interpolated>, bool> simplifyOMPL(statespace::StateSpacePtr _stateSpace,
+                                                                    statespace::InterpolatorPtr _interpolator,
+                                                                    distance::DistanceMetricPtr _dmetric,
+                                                                    constraint::SampleablePtr _sampler,
+                                                                    constraint::TestablePtr _validityConstraint,
+                                                                    constraint::TestablePtr _boundsConstraint,
+                                                                    constraint::ProjectablePtr _boundsProjector, 
+                                                                    double _maxDistanceBtwValidityChecks,
+                                                                    double _timeout, size_t _maxEmptySteps,
+                                                                    trajectory::InterpolatedPtr _originalTraj)
+{
+
+  // Step 1: Generate the space information of OMPL type
+  auto si = getSpaceInformation(
+        _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+        std::move(_validityConstraint), std::move(_boundsConstraint),
+        std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+
+  // Get the state space
+  auto sspace = ompl_static_pointer_cast<GeometricStateSpace>(
+        si->getStateSpace());
+
+  // Step 2: Convert the AIKIDO trajectory to Geometric Path 
+  ::ompl::geometric::PathGeometric *path = new ::ompl::geometric::PathGeometric(si);
```

Comment:
This leaks memory. I don't think `path` needs to be heap allocated, so you could replace this with just:
```c++
::ompl::geometric::PathGeometric path { si };
```